src/foundry/client-esm/applications/api/handlebars-application.d.mts

```diff
@@ -4,7 +4,23 @@ import type ApplicationV2 from "./application.d.mts";
 /**
  * The mixed application class augmented with [Handlebars](https://handlebarsjs.com) template rendering behavior.
  */
-declare class HandlebarsApplication {
+declare class HandlebarsApplication<
+  // BaseClass is the class being mixed. This is given by `HandlebarsApplicationMixin`.
+  BaseClass extends ConstructorOf<ApplicationV2.Internal<any, any, any>>,
+  // These type parameters should _never_ be explicitly assigned to. They're
+  // simply a way to make types more readable so that their names show up in
+  // intellisense instead of a transformation of `BaseClass`.
+  out RenderOptions extends ApplicationV2.RenderOptions = BaseClass extends ConstructorOf<
+    ApplicationV2.Internal<any, infer RenderOptions, any>
+  >
+    ? RenderOptions
+    : never,
+  out RenderContext extends Record<string, unknown> = BaseClass extends ConstructorOf<
```

Comment:
I also think the comment does a good job explaining the situation here. However just to explain a second way just in case, I use these parameters to give it a name so that `RenderOptions` shows up instead of `BaseClass extends ConstructorOf<ApplicationV2.Internal<any, infer RenderOptions, any>> ? RenderOptions : never` in the intellisense.